server/modules/routing/readwritesplit/rwsplit_route_stmt.cc

```diff
@@ -1134,7 +1135,44 @@ bool handle_got_target(RWSplit *inst, RWSplitSession *rses,
     ss_dassert(target->session_command_count() == 0);
 
     mxs::Backend::response_type response = mxs::Backend::NO_RESPONSE;
+    MySQLProtocol *backend_proto = DCB_PROTOCOL(target->dcb(), MySQLProtocol);
```

Comment:
The routers must not modify the protocol structures directly. If required, move either the protocol functionality into readwritesplit or some of the readwritesplit functionality into the protocol. Moving this into the protocol module would allow other modules to use it as well.